api/src/main/resources/liquibase.xml

```diff
@@ -0,0 +1,320 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+	xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-2.0.xsd
+    http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-2.0.xsd">
+
+	<changeSet id="43889130-55c2-49e9-b72b-a9eae9a31da7" author="Partners In Health">
+		<preConditions onFail="MARK_RAN">
+			<not><tableExists tableName="idgen_identifier_source"/></not>
+		</preConditions>
+		<createTable tableName="idgen_identifier_source">
+			<column name="id" type="int(11)" autoIncrement="true">
+				<constraints primaryKey="true" nullable="false" unique="true"/>
+			</column>
+			<column name="uuid" type="char(38)">
+				<constraints nullable="false"/>
+			</column>
+			<column name="name" type="varchar(255)">
+				<constraints nullable="false"/>
+			</column>
+			<column name="description" type="varchar(1000)">
+				<constraints nullable="false"/>
+			</column>
+			<column name="identifier_type" type="int(11)" defaultValue="0">
+				<constraints nullable="false" unique="true" foreignKeyName="identifier_type for idgen_identifier_source" references="patient_identifier_type(patient_identifier_type_id)"/>
+			</column>
+			<column name="creator" type="int(11)" defaultValue="0">
+				<constraints nullable="false" unique="true" foreignKeyName="creator for idgen_identifier_source" references="users(user_id)"/>
```

Comment:
Are you sure that the creator should be unique? Meaning that one cannot create more than one row in this table?

Reply:
Remove the unique constraint would mean, the table cannot be indexed by the creator column as specified in the sqldiff

Reply:
Could you point me to where it is indexed by the creator column? As in which line number.

Reply:
https://github.com/openmrs/openmrs-module-idgen/blob/448573a0d360a6929893d89c062f9968e5fbcf0e/omod/src/main/resources/sqldiff.xml#L32

Reply:
Does indexing a table field require its values to be unique?